sys/cheri/cheric.h

```diff
@@ -337,6 +337,51 @@ __cheri_clear_low_ptr_bits(uintptr_t ptr, size_t bits_mask) {
 #define cheri_clear_low_ptr_bits(ptr, mask)                                    \
   __cheri_clear_low_ptr_bits((uintptr_t)(ptr), __static_assert_sensible_low_bits(mask))
 
+#if __has_feature(capabilities)
+#define	CHERI_REPRESENTABLE_LENGTH(len) \
+	__builtin_cheri_round_representable_length(len)
+#define	CHERI_REPRESENTABLE_ALIGNMENT_MASK(len) \
+	__builtin_cheri_representable_alignment_mask(len)
+
+/*
+ * TODO: avoid using these since count leading/trailing zeroes is expensive on
+ * BERI/CHERI
```

Comment:
This could be updated to say it's expensive on MIPS and RISCV64 (checked with compiler explorer). It's cheap on Morello